### PR TITLE
Add unicode emoji to SelectMenuOptionBuilder

### PR DESCRIPTION
**Pull request by Mysterypotatoguy**
## Checklist
- [x] I have tested this PR[^1].
- [x] I have read the [contributing guidelines](./CONTRIBUTING.md).

## Changelog
- Added missing builder methods for setting the `emoji` property to a unicode emoji on Select Menu Options

Closes: NaN

[^1]: At least started a running bot instance with your changes and triggered an event so your changed code gets executed.

### CODE REVIEW

File: javacord-api/src/main/java/org/javacord/api/entity/message/component/SelectMenuOption.java [hunk 1]
```diff
@@ -123,6 +123,24 @@ static SelectMenuOption create(String label, String value, String description, E
...
     }
 
+    /**
+     * Creates a new select menu option with the given values.
+     *
+     * @param label The label for the option.
+     * @param value The value for the option.
+     * @param description The description for the option.
+     * @param unicodeEmoji The emoji for the option as a unicode string.
+     * @return The created select menu option.
```
Comment (RealYusufIsmail) [inline javacord-api/src/main/java/org/javacord/api/entity/message/component/SelectMenuOption.java:133]:
```suggestion
     * @param label        The label for the option.
     * @param value        The value for the option.
     * @param description  The description for the option.
     * @param unicodeEmoji The emoji for the option as a unicode string.
     * @return The created select menu option.
```

File: javacord-api/src/main/java/org/javacord/api/entity/message/component/SelectMenuOption.java [hunk 2]
```diff
@@ -142,4 +160,25 @@ static SelectMenuOption create(String label, String value, String description, E
...
+    /**
+     * Creates a new select menu option with the given values.
+     *
+     * @param label The label for the option.
+     * @param value The value for the option.
+     * @param description The description for the option.
+     * @param unicodeEmoji The emoji for the option as a unicode string.
+     * @param isDefault If the option is the default option.
+     * @return The created select menu option.
+     */
```
Comment (RealYusufIsmail) [inline javacord-api/src/main/java/org/javacord/api/entity/message/component/SelectMenuOption.java:173]:
```suggestion
    /**
     * Creates a new select menu option with the given values.
     *
     * @param label        The label for the option.
     * @param value        The value for the option.
     * @param description  The description for the option.
     * @param unicodeEmoji The emoji for the option as a unicode string.
     * @param isDefault    If the option is the default option.
     * @return The created select menu option.
     */
```

**Review by RealYusufIsmail — CHANGES_REQUESTED** [review]

**Comment by RealYusufIsmail** [comment]
Apart from that LGTM

**Review by Saladoc — APPROVED** [review]
The javadoc formatting is not a requirement. Green light from my side.

**Review by KILLEliteMaste — APPROVED** [review]

**Pull request merged**